site/src/pages/tutorials/tutorial-blog-rest-step6.mdx

```diff
@@ -0,0 +1,152 @@
+# Implementing DELETE operation
+
+In this step, we'll write a method for deleting a blog post. The HTTP method we map with this method is DELETE.
+
+<TutorialSteps current={6} />
+
+By completing this step, you'll learn to map your service with the HTTP DELETE (<type://@Delete>) method, customize an exception handler, and use a blocking task executor.
+
+## What you need
+
+You need to have these files prepared from previous steps.
+
+- [BlogService.java](/tutorials/tutorial-blog-rest-step3#0-create-a-service-file)
+- [BlogPost.java](/tutorials/tutorial-blog-rest-step2)
+
+## 1. Map HTTP method
+
+Let's start with [mapping the HTTP DELETE method](/docs/server-annotated-service#mapping-http-service-methods) with our service method:
+
+1. [Create a service file](/tutorials/tutorial-blog-rest-step3#0-create-a-service-file) if you're yet to create one.
+2. Declare a service method, `deleteBlogPost()` in the class `BlogService`.
+3. Map this service method with the HTTP DELETE method by adding the <type://@Delete> annotation.
+4. Bind the endpoint `/blogs` to the method.
+
+```java filename=BlogService.java highlight=1
+@Delete("/blogs")
+public HttpResponse deleteBlogPost(int id) {
+  // Implementation
+}
+```
+
+## 2. Handle parameters
+
+Let's take the blog post ID (`id`) as a path parameter for identifying the post to delete.
+
+1. Take in the ID value as a path parameter by adding `/blogs/:id` to the <type://@Delete> annotation.
+2. [Inject the path parameter](/docs/server-annotated-service#parameter-injection) to the service method by annotating the parameter with <type://@Param>.
+  ```java filename=BlogService.java
+  @Delete("/blogs/:id")
+  public HttpResponse deleteBlogPost(@Param int id) {
+    // Implementation
+  }
+  ```
+
+## 3. Implement service code
+
+In this step, write the code to delete a blog post, handle an exception, and block the operation.
+
+- [Delete a blog post](#delete-a-blog-post)
+- [Handle exceptions](#handle-exceptions)
+- [Add blocking](#add-blocking)
+
+### Delete a blog post
+
+Deleting a given blog post in this tutorial means removing a blog post from the map, `blogPosts`. However, in real services you would be performing this action on a database.
+
+To delete a blog post, copy the code below to the `deleteBlogPost()` method.
+
+```java filename=BlogService.Java highlight=2
+public HttpResponse deleteBlogPost(@Param int id) {
+  BlogPost removed = blogPosts.remove(id);
+  ...
+```
+
+### Handle exceptions
+
+What if there is no such post to delete? We can check if the blog exists before attempting to remove the blog post, but here, let's handle it after the attempt.
+
+1. Throw an `IllegalArgumentException` when no blog post exists with a given ID.
+  ```java filename=BlogService.java
+  if (removed == null) {
+    throw new IllegalArgumentException("The blog post does not exist. id: " + id);
+  }
+  ```
+2. Create an exception handler for the blog service:
+  1. Create a file, BadRequestExceptionHandler.java
```

Comment:
![image](https://user-images.githubusercontent.com/25103250/122151581-c62c2780-ce9a-11eb-9198-1293f86e826f.png)
This does not render properly. We need an extra space. 
```suggestion
   1. Create a file, BadRequestExceptionHandler.java
```

Reply:
Had been added along with adding import statements, 48b7d95931408b1e9f0d0fb877597525ffb443f8